src/hotspot/share/oops/markWord.hpp

```diff
@@ -107,12 +111,23 @@ class markWord {
   static const int self_forwarded_bits            = 1;
   static const int max_hash_bits                  = BitsPerWord - age_bits - lock_bits - self_forwarded_bits;
   static const int hash_bits                      = max_hash_bits > 31 ? 31 : max_hash_bits;
+  static const int hash_bits_compact              = max_hash_bits > 25 ? 25 : max_hash_bits;
```

Comment:
Maybe these constants should be `private` and be accessed through accessors instead.

Reply:
Good suggestion.  Also have an accessor function so that the if (UseCompactHeaders) isn't exposed in the callers.

Reply:
No wait, why? Those are constants. What do we win if we make accessors around them?

Reply:
What I mean is that we have an accessor like:

    static int hash_bits() {
        return UseCompactObjectHeaders ? _hash_bits_compact : _hash_bits_non_compact;
    }

Reply:
Right! That is a good suggestion! I'll do that in the next updated. Thank you!